src/services/configService.js

```diff
@@ -69,4 +70,22 @@ export class ConfigService {
     return HttpClient.put(`${ENDPOINT}solution-settings/logo`, logo, options)
       .map(prepareLogoResponse);
   }
+
+  /* get solution settings */
+  static getSolutionSettings() {
+    return HttpClient
+      .get(`${ENDPOINT}solution-settings/theme`)
+      .catch(error =>
+        error.status === 404 ? Observable.of(true) : Observable.throw(error)
+      );
+  }
+
+  /* update solution settings */
+  static updateSolutionSettings(model) {
+    return HttpClient
+      .put(`${ENDPOINT}solution-settings/theme`, model)
+      .catch(error =>
+        error.status === 404 ? Observable.of(true) : Observable.throw(error)
```

Comment:
Does 404 for the put call also mean the same thing as for the get?

Reply:
The scenario I told about you empty collection is primarily for GET call. PUT shouldn't return a 404 not found. It's kind of a upsert.